### PR TITLE
Review Member_part

### PR DESCRIPTION
**Pull request by VladPulyak**
I made Member_part in DIMS project

### CODE REVIEW

File: DIMS-Core.Database/DIMS-Core.Database/Functions/SetUSerTaskAsFail.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+create or alter procedure SetUserTaskAsFail @UserId int, @TaskId int 
+as
+BEGIN
+    update UserTask
+    set StateId = 3
+    where UserTaskId = 
+    (select UserTaskId from UserTask
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Functions/SetUSerTaskAsFail.sql]: You don't need this subquery. Filtration will be enoght.

File: DIMS-Core.Database/DIMS-Core.Database/Functions/SetUserTaskAsSuccess.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+create or alter procedure SetUserTaskAsSuccess @UserId int, @TaskId int 
+as
+BEGIN
+    update UserTask
+    set StateId = 2
+    where UserTaskId = 
+    (select UserTaskId from UserTask
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Functions/SetUserTaskAsSuccess.sql]: You don't need this subquery. Filtration will be enoght.

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+create table TaskState
+(
+    StateId int identity(1,1) not null,
+    StateName nvarchar(50) not null,
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql]: Make StateName unique

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
+create table TaskTrack
+(
+    TaskTrackId int identity(1,1) not null,
+    UserTaskId int not null,
+    TrackDate datetime not null,
+    TrackNote nvarchar(250) not null,
+    constraint PK_TaskTrackId primary key (TaskTrackId),
+    constraint FK_UserTaskId foreign key (UserTaskId) references UserTask(UserTaskId)
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql]: Add cascade on delete and on update

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+create table UserTask
+(
+    UserTaskId int identity(1,1) not null,
+    TaskId int not null,
+    UserId int not null,
+    StateId int not null,
+    constraint PK_UserTaskId primary key (UserTaskId),
+    constraint FK_TaskId foreign key (TaskId) references Task(TaskId),
+    constraint FK_UserId foreign key (UserId) references UserProfiles(UserId),
+    constraint FK_StateId foreign key (StateId) references TaskState(StateId)
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql]: Add cascade on delete and on update

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+create or alter view vUserTask
+as
+select UserProfiles.UserId, Task.TaskId, Task.Name as TaskName, Task.Description, Task.StartDate, Task.DeadlineDate, TaskState.StateName
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql]: You can user UserId column from UserTask table so you don't have to join UserProfiles table

File: DIMS-Core.Database/DIMS-Core.Database/SeedData/UserTrackSeedData.sql [hunk 7]
```diff
@@ -0,0 +1,11 @@
+insert into TaskTrack (UserTaskId, TrackDate, TrackNote) values
```
Comment (dbogdan2000) [inline DIMS-Core.Database/DIMS-Core.Database/SeedData/UserTrackSeedData.sql]: Why is the TaskTrack seed file named UserTrackSeedData?

**Review by dbogdan2000 — CHANGES_REQUESTED** [review]

File: DIMS-Core.Database/DIMS-Core.Database/Functions/SetUSerTaskAsFail.sql (hunk 1 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/Functions/SetUSerTaskAsFail.sql]: Fixed

File: DIMS-Core.Database/DIMS-Core.Database/Functions/SetUserTaskAsSuccess.sql (hunk 2 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/Functions/SetUserTaskAsSuccess.sql]: Fixed

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql (hunk 3 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskState.sql]: Fixed

File: DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql (hunk 4 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/TaskTrack.sql]: Resolved

File: DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql (hunk 5 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/Tables/UserTask.sql]: Fixed

File: DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql (hunk 6 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/Views/vUserTask.sql]: Resolved

File: DIMS-Core.Database/DIMS-Core.Database/SeedData/UserTrackSeedData.sql (hunk 7 above)
Comment (VladPulyak) [reply DIMS-Core.Database/DIMS-Core.Database/SeedData/UserTrackSeedData.sql]: Fixed

**Review by dbogdan2000 — APPROVED** [review]

**Pull request merged**